tests/test_cli/test_stack.py

```diff
@@ -0,0 +1,62 @@
+"""Test suit to test the stack command and all its subcommands."""
+from unittest.mock import patch
+
+from typer.testing import CliRunner
+
+from matcha_ml.cli.cli import app
+
+INTERNAL_FUNCTION_STUB = "matcha_ml.core"
+
+
+def test_cli_stack_command_help_option(runner: CliRunner) -> None:
+    """Tests the --help option for the cli stack command.
+
+    Args:
+        runner (CliRunner): typer CLI runner
+    """
+    result = runner.invoke(app, ["stack", "--help"])
+
+    assert result.exit_code == 0
+
+    assert "Configure the stack for Matcha to provision." in result.stdout
+
+
+def test_cli_stack_command_defaults_to_help(runner: CliRunner) -> None:
+    """Tests the --help option for the cli stack command.
+
+    Args:
+        runner (CliRunner): typer CLI runner
+    """
+    result = runner.invoke(app, ["stack"])
+
+    assert result.exit_code == 0
+
+    assert "Configure the stack for Matcha to provision." in result.stdout
+
+
+def test_cli_stack_set_command_help_option(runner: CliRunner) -> None:
+    """Tests the --help option for the cli stack set sub-command.
+
+    Args:
+        runner (CliRunner): typer CLI runner
+    """
+    result = runner.invoke(app, ["stack", "set", "--help"])
+
+    assert result.exit_code == 0
+
+    assert "Define the stack for Matcha to provision." in result.stdout
+
+
+def test_cli_stack_set_command(runner: CliRunner) -> None:
+    """Tests the cli stack set sub-command.
+
+    Args:
+        runner (CliRunner): typer CLI runner
+    """
+    with patch(f"{INTERNAL_FUNCTION_STUB}.stack_set") as mocked_stack_set:
```

Comment:
You probably don't need to mock this if you make use of the returned value from the `stack_set` function

Reply:
I didn't want to commit to anything being returned from `stack_set` as I didn't know what was in mind for that implementation. Mocking the placeholder function seemed sensible as it'll allow greater flexibility in the `stack_set` implementation without rewriting the tests.

Happy to rewrite it thought if you'd like?

Reply:
Yep, that's completely reasonable.

The underlying function is going to return something as an indicator, it's unlikely to be a `str` (as that's ambiguous; mostly likely an `enum` instance), but it's fine to make that assumption here. The tests can be updated at a later date, once the actual functionality is implemented. So I'd go without mocking.